cypress/utils/dashboards/data_explorer/commands.js

```diff
@@ -0,0 +1,154 @@
+});
+
+/**
+ * Get the Search Bar Date Picker End Date button.
+ */
+Cypress.Commands.add('getSearchDatePickerEndDateButton', () => {
+  return cy.getElementByTestId(DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_PICKER_END_DATE_BUTTON);
+});
+
+/**
+ * Get the Relative Date tab in the Search Bar Date Picker.
+ */
+Cypress.Commands.add('getDatePickerRelativeTab', () => {
+  return cy.getElementByTestId(DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_PICKER_RELATIVE_TAB);
+});
+
+/**
+ * Get the Relative Date Input in the Search Bar Date Picker.
+ */
+Cypress.Commands.add('getDatePickerRelativeInput', () => {
+  return cy.getElementByTestId(DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_RELATIVE_PICKER_INPUT);
+});
+
+/**
+ * Get the Absolute Date Input in the Search Bar Date Picker.
+ */
+Cypress.Commands.add('getDatePickerAbsoluteInput', () => {
+  return cy.getElementByTestId(DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_PICKER_ABSOLUTE_DATE_INPUT);
+});
+
+/**
+ * Get the Relative Date Unit selector in the Search Bar Date Picker.
+ */
+Cypress.Commands.add('getDatePickerRelativeUnitSelector', () => {
+  return cy.getElementByTestId(
+    DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_RELATIVE_PICKER_UNIT_SELECTOR
+  );
+});
+
+/**
+ * Get the Absolute Date tab in the Search Bar Date Picker.
+ */
+Cypress.Commands.add('getDatePickerAbsoluteTab', () => {
+  return cy.getElementByTestId(DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_PICKER_ABSOLUTE_TAB);
+});
+
+/**
+ * Get the Absolute Date Input in the Absolute Tab of the Search Bar Date Picker.
+ */
+Cypress.Commands.add('getDatePickerAbsoluteDateInput', () => {
+  return cy.getElementByTestId(DATA_EXPLORER_PAGE_ELEMENTS.SEARCH_DATE_PICKER_ABSOLUTE_DATE_INPUT);
+});
+
+/**
+ * Set search to start from a relative Date
+ * @param relativeNumber Relative integer string to set date range
+ * @param relativeUnit Unit for number. Accepted Units: seconds/Minutes/Hours/Days/Weeks/Months/Years ago/from now
+ * @example
+ * // sets search to return results from 15 years ago to now
+ * setSearchRelativeDateRange('15', 'years ago')
+ */
+Cypress.Commands.add('setSearchRelativeDateRange', (relativeNumber, relativeUnit) => {
+  cy.getSearchDatePickerButton().click();
+  cy.getDatePickerRelativeTab().click();
+  cy.getDatePickerRelativeInput().clear().type(relativeNumber);
+  cy.getDatePickerRelativeUnitSelector().select(relativeUnit);
+  cy.getQuerySubmitButton().click();
+});
+
+/**
+ * Set search to an absolute Date using searchDatePickerButton (this button only appears if the start date is "now" and end date is a relative date, or vice versa)
+ * @param absoluteStartDate String for Absolute Datetime for start date in format 'MMM dd, yyyy @ HH:mm:ss.SSS'.
+ * @param absoluteEndDate String for Absolute Datetime for end date in format 'MMM dd, yyyy @ HH:mm:ss.SSS'.
+ * @example setSearchAbsoluteDateRangeWithSearchDatePickerButton('Dec 31, 2020 @ 16:00:00.000', 'Dec 31, 2022 @ 14:14:42.801')
+ */
+Cypress.Commands.add(
```

Comment:
Is this to set absolute time in the time range? I think the current method can set absolute time range right? why create a new one?
```
Cypress.Commands.add("setTopNavDate", (start, end, submit = true) => {
  const opts = { log: false };

  Cypress.log({
    name: "setTopNavDate",
    displayName: "set date",
    message: `Start: ${start} :: End: ${end}`,
  });

  /* Find any one of the two buttons that change/open the date picker:
   *   * if `superDatePickerShowDatesButton` is found, it will switch the mode to dates
   *      * in some versions of OUI, the switch will open the date selection dialog as well
   *   * if `superDatePickerstartDatePopoverButton` is found, it will open the date selection dialog
   */
  cy.getElementsByTestIds(
    ["superDatePickerstartDatePopoverButton", "superDatePickerShowDatesButton"],
    opts
  )
    .should("be.visible")
    .invoke("attr", "data-test-subj")
    .then((testId) => {
      cy.getElementByTestId(testId, opts).should("be.visible").click(opts);
    });

  /* While we surely are in the date selection mode, we don't know if the date selection dialog
   * is open or not. Looking for a tab and if it is missing, click on the dialog opener.
   */
  cy.whenTestIdNotFound("superDatePickerAbsoluteTab", () => {
    cy.getElementByTestId("superDatePickerstartDatePopoverButton", opts)
      .should("be.visible")
      .click(opts);
  });

  // Click absolute tab
  cy.getElementByTestId("superDatePickerAbsoluteTab", opts).click(opts);

  // Type absolute start date
  cy.getElementByTestId("superDatePickerAbsoluteDateInput", opts)
    .click(opts)
    .clear(opts)
    .type(start, {
      ...opts,
      delay: 0, // add a delay here, cypress sometimes fails to type all the content into the input.
    });

  // Click end date
  cy.getElementByTestId("superDatePickerendDatePopoverButton", opts)
    .last(opts)
    .click(opts);

  // Click absolute tab
  cy.getElementByTestId("superDatePickerAbsoluteTab", opts)
    .last(opts)
    .click(opts);

  // Type absolute end date
  cy.getElementByTestId("superDatePickerAbsoluteDateInput", opts)
    .last(opts)
    .click(opts)
    .clear(opts)
    .type(end, {
      ...opts,
      delay: 0, // add a delay here, cypress sometimes fails to type all the content into the input.
    });

  // Close popup
  cy.getElementByTestId("superDatePickerendDatePopoverButton", opts).click(
    opts
  );

  if (submit) {
    cy.updateTopNav(opts);
  }
});
```

Reply:
This problem only surfaced due to workspaces caching the absolute start and end dates. I didn't realise that Justin's merged pr already had that solution. Will be replacing.